qiita_db/support_files/patches/python_patches/25.py

```diff
@@ -0,0 +1,121 @@
+# May 19, 2015
+# We attach the prep template directly to the study. The raw data is no longer
+# attached to the study directly, the prep template point to them. This will
+# make the RawData to be effectively just a container for the raw files,
+# which is how it was acting previously.
+
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.data import RawData
+from qiita_db.util import move_filepaths_to_upload_folder
+
+conn_handler = SQLConnectionHandler()
+queue = "PATCH_25"
+conn_handler.create_queue(queue)
+
+# the system may contain raw data with no prep template associated to it.
+# Retrieve all those raw data ids
+sql = """SELECT raw_data_id
+         FROM qiita.raw_data
+         WHERE raw_data_id NOT IN (
+            SELECT DISTINCT raw_data_id FROM qiita.prep_template);"""
+rd_ids = [x[0] for x in conn_handler.execute_fetchall(sql)]
+
+# We will delete those RawData. However, if they have files attached, we should
+# move them to the uploads folder of the study
+sql_detach = """DELETE FROM qiita.study_raw_data
```

Comment:
The reason is that the patch will be executed with the new code. In the new code, the RawData does not know to which studies it belongs unless it is connected to a PrepTemplate. The RawData objects that are here, are "duplicating" the existing code, but once all the pieces are in place, that code is different. Does this make sense to you?